Factory/EventDispatcherFactory.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\NotificationBundle\Factory;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\ContainerAwareEventDispatcher;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+use Symfony\Component\HttpKernel\Kernel;
+
+class EventDispatcherFactory
```

Comment:
Looks like that to me:
https://github.com/symfony/symfony/blob/3.2/src/Symfony/Bundle/FrameworkBundle/Resources/config/services.xml <-- Services in 3.2

https://github.com/symfony/symfony/blob/3.3/src/Symfony/Bundle/FrameworkBundle/Resources/config/services.xml <-- Services in 3.3

Both have event_dispatcher  as ContainerAwareEventDispatcher